controllers/noobaa-rules.yaml

```diff
@@ -1,76 +1,65 @@
+# NOTE: Try to keep the cardinality of the rules as low as possible,
+# while reducing the computation cost on the PromQL side as much as possible.
  groups:
    - name: noobaa-telemeter.rules
      rules:
        - expr: |
-           NooBaa_num_buckets_claims{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_buckets_claims{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_num_buckets_claims
+         record: namespace:noobaa_buckets_claims:max
        - expr: |
-           NooBaa_num_unhealthy_bucket_claims{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_unhealthy_bucket_claims{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_num_unhealthy_bucket_claims
+         record: namespace:noobaa_unhealthy_bucket_claims:max
        - expr: |
-           NooBaa_num_unhealthy_namespace_buckets{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_unhealthy_namespace_buckets{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: job:noobaa_total_unhealthy_namespace_buckets
+         record: namespace:noobaa_unhealthy_namespace_buckets:max
        - expr: |
-           NooBaa_num_namespace_buckets{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_accounts_num{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: job:noobaa_namespace_bucket_count
+         record: namespace:noobaa_accounts:max
        - expr: |
-           NooBaa_accounts_num{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_total_usage{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_accounts_num
+         record: namespace:noobaa_usage:max
        - expr: |
-           NooBaa_total_usage{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_unhealthy_namespace_resources{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_total_usage
+         record: namespace:noobaa_unhealthy_namespace_resources:max
        - expr: |
-            NooBaa_num_unhealthy_namespace_resources{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_namespace_resources{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_num_unhealthy_namespace_resources
+         record: namespace:noobaa_namespace_resources:max
        - expr: |
-           NooBaa_num_namespace_resources{namespace="redhat-data-federation"}
+           max(NooBaa_num_namespace_buckets{namespace="redhat-data-federation"})
```

Comment:
`namespace:noobaa_unhealthy_namespace_buckets:max` is `max without(instance) (...)` so the label set won't match between the 2 rules leading to more complexity when comparing both metrics.
It might also make sense to reorder the rules list to have related rules close to each other (e.g. `namespace:noobaa_namespace_buckets:max` and `namespace:noobaa_unhealthy_namespace_resources:max` together)

Reply:
Makes sense, refactored the rule file, PTAL.